src/sources/docker.rs

```diff
@@ -373,11 +247,149 @@ fn docker_source(
     Ok(main)
 }
 
+fn docker_event_stream(
+    config: &DockerConfig,
+) -> Result<
+    impl Stream<Item = async_docker::Result<async_docker::Event>, Error = async_docker::Error>,
```

Comment:
First Result is necessary because Docker construction can fail. 
Second Result is part of signature of Stream that async_docker::DockerApi::events returns.

Or do you see something else?

Reply:
A, I see now, with shiplift, it's.

Reply:
`shiplift` probably defers construction failure to later calls, so its Docker construction can't fail, or at least it's not shown at that point.